examples/misc/upload_image_assets.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This code example uploads an image asset.
+
+To get image assets, run get_all_image_assets.py."""
+
+
+import argparse
+import sys
+import requests
+
+from google.ads.google_ads.client import GoogleAdsClient
+from google.ads.google_ads.errors import GoogleAdsException
+
+
+def main(client, customer_id):
+    """Main method, to run this code example as a standalone application."""
+
+    # Download image from URL
+    URL = "https://goo.gl/3b9Wfh"
+    image_content = requests.get(URL).content
+
+    asset_operation = client.get_type('AssetOperation', version='v2')
+    asset = asset_operation.create
+    asset.type = client.get_type('AssetTypeEnum', version='v2').IMAGE
+    asset.image_asset.data.value = image_content
+    asset.image_asset.file_size.value = len(image_content)
+    asset.image_asset.mime_type = client.get_type('MimeTypeEnum').IMAGE_JPEG
+    # Use your favorite image library to determine dimensions
+    asset.image_asset.full_size.height_pixels.value = 315
```

Comment:
As an aside, I wonder if an interesting utility might be one that can identify the image type and size for a given file / URL. If all image assets across the API are represented in a similar way, it should be possible to even populate and return an image_asset that can be used for API calls such as this. Perhaps not in the scope of this change, however. :-)

Reply:
There are various libraries (e.g. PIL) and pure Python solutions, but adding them would increase dependencies or distract from this sample so I left it hardcoded for now as an exercise for the viewer. :)